src/stcal/outlier_detection/utils.py

```diff
@@ -0,0 +1,337 @@
+
+    pix_ratio : float
+        Pixel ratio.
+
+    Returns
+    -------
+    blotted : numpy.ndarray
+        The blotted median data.
+
+    blot_img : datamodel
+        Datamodel containing header and WCS to define the 'blotted' image
+    """
+    # Compute the mapping between the input and output pixel coordinates
+    pixmap = calc_gwcs_pixmap(blot_wcs, median_wcs, blot_shape)
+    log.debug("Pixmap shape: {}".format(pixmap[:, :, 0].shape))
+    log.debug("Sci shape: {}".format(blot_shape))
+    log.info('Blotting {} <-- {}'.format(blot_shape, median_data.shape))
+
+    outsci = np.zeros(blot_shape, dtype=np.float32)
+
+    # Currently tblot cannot handle nans in the pixmap, so we need to give some
+    # other value.  -1 is not optimal and may have side effects.  But this is
+    # what we've been doing up until now, so more investigation is needed
+    # before a change is made.  Preferably, fix tblot in drizzle.
+    pixmap[np.isnan(pixmap)] = -1
+    tblot(median_data, pixmap, outsci, scale=pix_ratio, kscale=1.0,
+          interp='linear', exptime=1.0, misval=0.0, sinscl=1.0)
+
+    return outsci
+
+
+def calc_gwcs_pixmap(in_wcs, out_wcs, in_shape):
+    """
+    Return a pixel grid map from input frame to output frame.
+
+    Parameters
+    ----------
+    in_wcs : gwcs.wcs.WCS
+        Input/source wcs.
+
+    out_wcs : gwcs.wcs.WCS
+        Output/projected wcs.
+
+    in_shape : list of int
+        Input shape used to compute the input bounding box.
+
+    Returns
+    -------
+    pixmap : numpy.ndarray
+        Computed pixmap.
+    """
+    bb = wcs_bbox_from_shape(in_shape)
+    log.debug("Bounding box from data shape: {}".format(bb))
+
+    grid = gwcs.wcstools.grid_from_bounding_box(bb)
+    return np.dstack(reproject(in_wcs, out_wcs)(grid[0], grid[1]))
+
+
+def reproject(wcs1, wcs2):
+    """
+    Given two WCSs return a function which takes pixel
+    coordinates in wcs1 and computes them in wcs2.
+
+    It performs the forward transformation of ``wcs1`` followed by the
+    inverse of ``wcs2``.
+
+    Parameters
+    ----------
+    wcs1, wcs2 : gwcs.wcs.WCS
+        WCS objects that have `pixel_to_world_values` and `world_to_pixel_values`
+        methods.
+
+    Returns
+    -------
+    _reproject :
+        Function to compute the transformations.  It takes x, y
+        positions in ``wcs1`` and returns x, y positions in ``wcs2``.
+    """
+
+    try:
+        forward_transform = wcs1.pixel_to_world_values
+        backward_transform = wcs2.world_to_pixel_values
+    except AttributeError as err:
+        raise TypeError("Input should be a WCS") from err
+
+    def _reproject(x, y):
+        sky = forward_transform(x, y)
+        flat_sky = []
+        for axis in sky:
+            flat_sky.append(axis.flatten())
+        # Filter out RuntimeWarnings due to computed NaNs in the WCS
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", RuntimeWarning)
```

Comment:
at some point I updated all the warnings in the `jwst` repo to use `warnings.filterwarnings(whatever, whatever, message="specific message")` to be absolutely sure we don't accidentally suppress unintended warnings.  Could you do the same here?

Reply:
Good suggestion.
I don't see a message for the copied jwst code:
https://github.com/spacetelescope/jwst/blob/bc255f4cffa7a887ab3d1eed7ebfe82b856a4c67/jwst/resample/resample_utils.py#L158-L159
and running this (with the test data) doesn't generate any warnings. @nden any tips on how to trigger a warning here to see the message?

Reply:
I'm wondering if we could just remove the warning filtering. It was added in:
https://github.com/spacetelescope/jwst/pull/3785
and I can't seem to reproduce those issues (I did not try running the old jwst code, just setting up transforms with nans and performing comparisons against nans). I wonder if numpy no longer emits these warnings (I tried back to numpy 1.21 and I don't get a warning using `nan` in a less operation).

Reply:
Thanks for trying these things. I agree it can just be removed if you aren't able to reproduce it.